### PR TITLE
Upgrade cloudevent to v2

### PR DESCRIPTION
**Pull request by homily707**
**What this PR does / why we need it**:
upgrde cloudevents  go-sdk in logger to v2  https://github.com/cloudevents/sdk-go

**Type of changes**
Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by homily707** [comment]
@lizzzcai @theofpa  review requested

**Comment by homily707** [comment]
@yuzisun  do I need to rebase master？ i see you merge master into this branch, and i'm not sure what to do

File: pkg/logger/handler.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func (eh *LoggerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			Url:              eh.logUrl,
 			Bytes:            &body,
 			ContentType:      contentType,
-			ReqType:          InferenceRequest,
+			ReqType:          CEInferenceRequest,
```
Comment (yuzisun) [inline pkg/logger/handler.go:94]: this changes from `Request` to CEInferenceRequest, does it break anything?

Comment (yuzisun) [reply pkg/logger/handler.go:94]: nvm I see it is setting to CEInferenceRequest later on, so we can remove InferenceRequest type

**Pull request merged**